### DISCUSSION TITLE
ProfilingATNSimulator implement

### DISCUSSION THREAD

**Discussion by backsapce** [mike-lischke/antlr4ng#19] [General]
recently, i porting my grammar to typescript runtime, antlr4ng missing ProfilingATNSimulator now,it is important analysis tool to improve parsing performence in my workflow. i had implement a beta version but missing test cases, The trouble is change some function signature in Parser.i'l make a pr soon  if i'm ready.

**Comment by mike-lischke** [comment]
Right, the current antlr4ng package is not complete. I converted the JS runtime to TS and added what I needed immediately. I'm currently working on the runtime tests from ANTLR4. Once this is available in TS I want to add remaining and missing stuff to antlr4ng and do some refactoring. Any help with that is highly welcome!

The current runtime tests (in Java) entirely succeed, which means there are no test for anything that is added later. Hence it is very helpful to have new test cases for that. I have not yet though about how to deal with these new tests later (e.g. integrate them with the runtime tests), however, the general idea is that each runtime also maintains its own tests and can use their own test environment.
